src/ck-ldb/TreeLevel.h

```diff
@@ -0,0 +1,1236 @@
+#ifndef TREELEVEL_H
+#define TREELEVEL_H
+
+#include "TopoManager.h"
+#include "TreeLB.h"
+#include "TreeStrategyBase.h"
+#include "TreeStrategyFactory.h"
+#include <cmath>
+#include <limits>  // std::numeric_limits
+
+#define FLOAT_TO_INT_MULT 10000
+
+// TODO rateAware is not implemented, that is: collecting processor speeds
```

Comment:
Is this true? Isn't that a step backwards?

Reply:
Fixed in #2871.